Clarify the Mod Tap defines in documentation (#4352)
qmk/qmk_firmware: docs/feature_advanced_keycodes.md

For Instance:

- `SHFT_T(KC_A)` Down
- `KC_X` Down
- `KC_X` Up
- `SHFT_T(KC_A)` Up

Normally, if you do all this within the `TAPPING_TERM` (default: 200ms) this will be registered as `ax` by the firmware and host system. With permissive hold enabled, this modifies how this is handled by considering the Mod Tap keys as a Mod if another key is tapped, and would registered as `X` (`SHIFT`+`x`). 

?> If you have `Ignore Mod Tap Interrupt` enabled, as well, this will modify how both work. The regular key has the modifier added if the first key is released first or if both keys are held longer than the `TAPPING_TERM`.

# Ignore Mod Tap Interrupt

To enable this setting, add this to your `config.h`:

```c
#define IGNORE_MOD_TAP_INTERRUPT
```

Similar to Permissive Hold, this alters how the firmware processes input for fast typist. If you press a Mod Tap key, press another key, release the Mod Tap key, and then release the normal key, it would normally output the "tapping" function for both keys. This may not be desirable for rolling combo keys. 

> This setting makes it so that you have to hold both keys for the `TAPPING_TERM` to trigger the hold function (the mod).

Reviewer: ```suggestion
Setting `Ignore Mod Tap Interrupt` requires  holding both keys for the `TAPPING_TERM` to trigger the hold function (the mod).
```

For Instance:

- `SHFT_T(KC_A)` Down
- `KC_X` Down
- `SHFT_T(KC_A)` Up
- `KC_X` Up

> Normally, this would sent `X` (`SHIFT`+`x`). With `Ignore Mod Tap Interrupt` enabled, it requires you to hold both keys for the `TAPPING_TERM` to register the hold action, and a quick tap will output `ax` in this case, while a hold on both will still output `X`  (`SHIFT`+`x`).

Reviewer: ```suggestion
Normally, this would send `X` (`SHIFT`+`x`). With `Ignore Mod Tap Interrupt` enabled, holding both keys are required for the `TAPPING_TERM` to register the hold action. A quick tap will output `ax` in this case, while a hold on both will still output `X`  (`SHIFT`+`x`).
```

?> __Note__: This only concerns modifiers and not layer switching keys.

?> If you have `Permissive Hold` enabled, as well, this will modify how both work. The regular key has the modifier added if the first key is released first or if both keys are held longer than the `TAPPING_TERM`.

# Tapping Force Hold

> To enable this, add this to your `config.h`: 

Reviewer: ```suggestion
To enable tapping force hold, add the following to your `config.h`: 
```

# Retro Tapping

> To enable this, add this to your `config.h`: 

Reviewer: ```suggestion
To enable retro tapping, add the following to your `config.h`: 
```
